src/main/java/com/fpl/smdc/core/rest/handler/HttpClientConfig.java

```diff
@@ -0,0 +1,190 @@
+
+  /** The Constant CLOSE_IDLE_CONNECTION_WAIT_TIME_SECS. */
+  private static final int CLOSE_IDLE_CONNECTION_WAIT_TIME_SECS = 30;
+
+  /**
+   * Connection keep alive strategy.
+   *
+   * @return the connection keep alive strategy
+   */
+  @Bean
+  public ConnectionKeepAliveStrategy connectionKeepAliveStrategy() {
+    return new ConnectionKeepAliveStrategy() {
+      @Override
+      public long getKeepAliveDuration(HttpResponse response, HttpContext context) {
+        HeaderElementIterator it =
+            new BasicHeaderElementIterator(response.headerIterator(HTTP.CONN_KEEP_ALIVE));
+        while (it.hasNext()) {
+          HeaderElement he = it.nextElement();
+          String param = he.getName();
+          String value = he.getValue();
+
+          if (value != null && param.equalsIgnoreCase("timeout")) {
+            return Long.parseLong(value) * 1000;
+          }
+        }
+        return DEFAULT_KEEP_ALIVE_TIME_MILLIS;
+      }
+    };
+  }
+
+  /**
+   * Http client.
+   *
+   * @return the closeable http client
+   */
+  @Bean
+  public CloseableHttpClient httpClient() {
+    RequestConfig requestConfig =
+        RequestConfig.custom().setConnectionRequestTimeout(REQUEST_TIMEOUT)
+            .setConnectTimeout(CONNECT_TIMEOUT).setSocketTimeout(SOCKET_TIMEOUT).build();
+
+    return HttpClients.custom()// .setSSLHostnameVerifier(new NoopHostnameVerifier())
+        .setDefaultRequestConfig(requestConfig).setConnectionManager(poolingConnectionManager())
+        .setKeepAliveStrategy(connectionKeepAliveStrategy()).build();
+  }
+
+  /**
+   * Idle connection monitor.
+   *
+   * @param connectionManager the connection manager
+   * @return the runnable
+   */
+  @Bean
+  public Runnable idleConnectionMonitor(
+      final PoolingHttpClientConnectionManager connectionManager) {
+    return new Runnable() {
+      @Override
+      @Scheduled(fixedDelay = 10000)
+      public void run() {
+        try {
+          if (connectionManager != null) {
+            LOGGER.trace("run IdleConnectionMonitor - Closing expired and idle connections...");
+            connectionManager.closeExpiredConnections();
+            connectionManager.closeIdleConnections(CLOSE_IDLE_CONNECTION_WAIT_TIME_SECS,
+                TimeUnit.SECONDS);
+          } else {
+            LOGGER.trace(
+                "run IdleConnectionMonitor - Http Client Connection manager is not initialised");
+          }
+        } catch (Exception e) {
+          LOGGER.error("run IdleConnectionMonitor - Exception occurred. msg={}, e={}",
+              e.getMessage(), e);
+        }
+      }
+    };
+  }
+
+  /**
+   * Pooling connection manager.
+   *
+   * @return the pooling http client connection manager
+   */
+  @Bean
+  public PoolingHttpClientConnectionManager poolingConnectionManager() {
+    TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
+
+    FileInputStream inputStream = null;
+    try {
+      inputStream =

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *inputStream*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by KeyStore.getInstance. Either a) close *inputStream* in a try-finally block or b) close the resource by declaring *inputStream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).